airbyte-webapp/src/components/connectorBuilder/Builder/Builder.tsx

```diff
@@ -0,0 +1,45 @@
+import { Form } from "formik";
+import { useEffect, useState } from "react";
+
+import { useConnectorBuilderState } from "services/connectorBuilder/ConnectorBuilderStateService";
+
+import { BuilderFormValues } from "../types";
+import styles from "./Builder.module.scss";
+import { BuilderSidebar, BuilderView } from "./BuilderSidebar";
+import { GlobalConfigView } from "./GlobalConfigView";
+import { StreamConfigView } from "./StreamConfigView";
+
+interface BuilderProps {
+  values: BuilderFormValues;
+  toggleYamlEditor: () => void;
+}
+
+export const Builder: React.FC<BuilderProps> = ({ values, toggleYamlEditor }) => {
+  const { setBuilderFormValues, setSelectedStream } = useConnectorBuilderState();
+  useEffect(() => {
+    setBuilderFormValues(values);
+  }, [values, setBuilderFormValues]);
+
+  const [selectedView, setSelectedView] = useState<BuilderView>("global");
+
+  const handleViewSelect = (selectedView: BuilderView, streamName?: string) => {
+    setSelectedView(selectedView);
+    if (selectedView !== "global" && streamName !== undefined) {
+      setSelectedStream(streamName);
```

Comment:
We are auto-selecting the testing panel for the stream when switching the view, but we don't do it the other way around (switching the view when switching the `selectedStream`). Would it make sense to do so?

Related to that - would it make sense to lift the selected view into the ConnectorBuilderStateService as well? It's conceptually very similar to the selected stream, it confused me at first why it's handled in such a different fashion.

Not a must-have though.

Reply:
Good suggestion -- I've added this behavior. The one slightly nuanced aspect of it is that if the Global Configuration view is currently selected while the user changes streams in the testing panel dropdown, I don't think we want it to switch the view to that stream, because the user might want to test out different streams while iterating on global configuration values. So I've encoded that behavior as well.